pkg/sink/sink.go

```diff
@@ -0,0 +1,130 @@
+)
+
+type Resource struct {
+	TriggersClient         *triggersclientset.Clientset
+	KubeClient             *kubeclientset.Clientset
+	EventListenerName      string
+	EventListenerNamespace string
+}
+
+func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Request) {
+	el, err := r.TriggersClient.TektonV1alpha1().EventListeners(r.EventListenerNamespace).Get(r.EventListenerName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting EventListener %s in Namespace %s: %s", r.EventListenerName, r.EventListenerNamespace, err)
+		return
+	}
+
+	event, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		log.Printf("Error reading event body: %s", err)
+		return
+	}
+
+	for _, tbRef := range el.Spec.TriggerBindingRefs {
+		tbNamespace := tbRef.Namespace
+		if tbNamespace == "" {
+			tbNamespace = r.EventListenerNamespace
+		}
+		tb, err := r.TriggersClient.TektonV1alpha1().TriggerBindings(tbNamespace).Get(tbRef.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("Error getting TriggerBinding %s in Namespace %s: %s", tbRef.Name, tbNamespace, err)
+			continue
+		}
+
+		for _, templateBinding := range tb.Spec.TemplateBindings {
+			ttNamespace := templateBinding.TemplateRef.Namespace
+			if ttNamespace == "" {
+				ttNamespace = r.EventListenerNamespace
+			}
+			tt, err := r.TriggersClient.TektonV1alpha1().TriggerTemplates(ttNamespace).Get(templateBinding.TemplateRef.Name, metav1.GetOptions{})
+			if err != nil {
+				log.Printf("Error getting TriggerTemplate %s in Namespace %s: %s", templateBinding.TemplateRef.Name, ttNamespace, err)
+				continue
+			}
+
+			params, err := template.ApplyEventToParams(event, templateBinding.Params)
+			if err != nil {
+				log.Printf("Error applying event to TriggerBinding params: %s", err)
+				continue
+			}
+			params = template.AddDefaultParamsFromSpecs(params, tt.Spec.Params)
+
+			for _, rt := range tt.Spec.ResourceTemplates {
+				rtRaw := template.ApplyParamsToResourceTemplate(params, rt.RawMessage)
+				err = createResource(r.KubeClient, rtRaw)
+				if err != nil {
+					log.Printf("Error creating resources from resource template %s: %s", string(rtRaw), err)
+					continue
+				}
+				log.Printf("Successfully created resource: %s", string(rtRaw))
+			}
+		}
+	}
+}
+
+// createResource uses the kubeClient to create the resource defined in the
+// TriggerResourceTemplate and returns any errors with this process
+func createResource(kubeClient *kubeclientset.Clientset, rt json.RawMessage) error {
+	// Assume the TriggerResourceTemplate is valid (it has an apiVersion and Kind)
+	result := kubeClient.RESTClient().Post().
+		RequestURI(getRequestURI(rt)).
+		Body([]byte(rt)).
+		SetHeader("Content-Type", "application/json").
+		Do()
+	if result.Error() != nil {
+		return result.Error()
+	}
+	return nil
+}
+
+// getRequestURI returns the URI for a post request to the kubernetes API
+// REST endpoint to create the TriggerResourceTemplate
```

Comment:
nit: Shouldn't it be to `create the resource defined in the TriggerResourceTemplate `  instead of  `create the TriggerResourceTemplate`

Reply:
Thanks, good catch 🙂